### PR TITLE
feat: Build, package and install browser extension when recording is started

### PR DESCRIPTION
**Pull request by allansson**
## Description

This PR sets up the build pipeline for the browser extension that's going to be used to record browser events. To test that it works, it also loads the extension when a recording is started.

## How to Test

1. Run the app in dev mode: `npm start`.
2. Start a recording.
3. Check that the extension has been installed.
    Screenshot 2024-12-11 at 19 55 44
4. Click manage extensions.
5. Enable developer mode.
   Screenshot 2024-12-11 at 19 56 15
7. Click the extension and inspect the service worker:
    Screenshot 2024-12-11 at 19 59 49
    If the view says that the service worker is inactive, then you've been idle for too long and need close Chrome and restart the recording.
8. Check the console tab and see that a message has been logged.
9. Open `forge.config.ts` in an editor.
10. Comment out the `osxSign` and `osxNotarize` properties.
11. Run `npm run package`.
12. Open the executable in `out/`
13. Test steps 2-8

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [x] I have run linter locally (`npm run lint`) and all checks pass.
- [x] I have run tests locally (`npm test`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

https://github.com/grafana/k6-studio/issues/233

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   "scripts": {
     "preinstall": "node install-k6.js",
     "start": "electron-forge start",
+    "start:extension": "STANDALONE_EXTENSION=true vite --config vite.extension.config.mts",
```
Comment (allansson) [inline package.json:11]: This lets us start a browser with the extension installed without running k6 Studio. Handy when you're more focused on the extension side of things.

File: src/browser.ts [hunk 2]
```diff
@@ -69,6 +81,7 @@ export const launchBrowser = async (
       '--disable-search-engine-choice-screen',
       `--proxy-server=http://localhost:${appSettings.proxy.port}`,
       `--ignore-certificate-errors-spki-list=${certificateSPKI}`,
+      `--load-extension=${extensionPath}`,
```
Comment (allansson) [inline src/browser.ts:84]: This will be put behind a feature flag that can be toggled via settings in a future PR.

File: vite.base.config.ts [hunk 3]
```diff
@@ -28,6 +28,7 @@ export function getBuildConfig(env: ConfigEnv<'build'>): UserConfig {
       outDir: '.vite/build',
       watch: command === 'serve' ? {} : null,
       minify: command === 'build',
+      sourcemap: command === 'serve',
```
Comment (allansson) [inline vite.base.config.ts:31]: I needed to add this to get debugging the main process working.

**Review by allansson — COMMENTED** [review]
Self-review

**Review by e-fisher — APPROVED** [review]
LGTM 🚀 
Changed base branch by accident, reverted it!

**Review by going-confetti — APPROVED** [review]
LGTM

**Pull request merged**